router/pkg/mcpserver/server.go

```diff
@@ -0,0 +1,676 @@
+		),
+		s.handleListGraphQLOperations(),
+	)
+
+	s.server.AddTool(
+		mcp.NewTool(
+			"get_operation_info",
+			mcp.WithDescription("Retrieve comprehensive metadata and execution details for a specific GraphQL operation by its name. Use this to collect all required information needed to execute the operation via execute_<operation_name>."),
+			mcp.WithString("operationName",
+				mcp.Required(),
+				mcp.Description("The exact name of the GraphQL operation to retrieve information for."),
+			),
+		),
+		s.handleGraphQLOperationInfo(),
+	)
+
+	// Only register the schema tool if exposeSchema is enabled
+	if s.exposeSchema {
+		s.server.AddTool(
+			mcp.NewTool(
+				"get_schema",
+				mcp.WithDescription("Use this function to obtain the full introspection-based schema of the GraphQL API. This is useful for understanding the structure, available types, queries, mutations, and overall capabilities of the API."),
+			),
+			s.handleGetGraphQLSchema(),
+		)
+	}
+
+	// Only register the execute_graphql tool if enableArbitraryOperations is enabled
+	if s.enableArbitraryOperations {
+		// Add a tool to execute arbitrary GraphQL queries
+		executeGraphQLSchema := []byte(`{
+			"type": "object",
+			"description": "The query and variables to execute.",
+			"properties": {
+				"query": {
+					"type": "string",
+					"description": "The GraphQL query or mutation string to execute."
+				},
+				"variables": {
+					"type": "object",
+					"additionalProperties": true,
+					"description": "The variables to pass to the GraphQL query as a JSON object."
+				}
+			},
+			"additionalProperties": false,
+			"required": ["query"]
+		}`)
+
+		s.server.AddTool(
+			mcp.NewToolWithRawSchema(
+				"execute",
+				"Executes a GraphQL query or mutation.",
+				executeGraphQLSchema,
+			),
+			s.handleExecuteGraphQL(),
+		)
+	}
+
+	// Get operations filtered by the excludeMutations setting
+	operations := s.operationsManager.GetFilteredOperations()
+
+	for _, op := range operations {
+		var compiledSchema *jsonschema.Schema
+		var err error
+
+		if len(op.JSONSchema) > 0 {
+			compiledSchema, err = s.schemaCompiler.CompileSchema(op)
+			if err != nil {
+				s.logger.Error("failed to compile schema for operation",
+					zap.String("operation", op.Name),
+					zap.Error(err))
+				continue
+			}
+		}
+
+		// Create handler with pre-compiled schema
+		handler := &operationHandler{
+			operation:      op,
+			compiledSchema: compiledSchema,
+		}
+
+		// Convert operation name to snake_case for consistent tool naming
+		toolName := toSnakeCase(op.Name)
+
+		var toolDescription string
+
+		if op.Description != "" {
+			toolDescription = fmt.Sprintf("Executes the GraphQL operation '%s' of type %s. %s", op.Name, op.OperationType, op.Description)
+		} else {
+			toolDescription = fmt.Sprintf("Executes the GraphQL operation '%s' of type %s.", op.Name, op.OperationType)
+		}
+
+		s.server.AddTool(
+			mcp.NewToolWithRawSchema(
+				fmt.Sprintf("execute_%s", toolName),
```

Comment:
maybe "execute_operation_%s"?

Reply:
I have to think and research a bit more about the naming scheme. A bit more concrete naming might be indeed better.